aws_embedded_metrics/logger/metrics_logger.py

```diff
@@ -73,6 +75,35 @@ def put_metric(self, key: str, value: float, unit: str = "None") -> "MetricsLogg
         self.context.put_metric(key, value, unit)
         return self
 
+    def add_stack_trace(self, key: str, value: Any = None, exc_info: Tuple = None) -> "MetricsLogger":
```

Comment:
What’s the use case for allowing a user provided exc_info?

Reply:
I've ended up in situations before where my context changes and the right thing isn't in sys.exc_info() anymore. More generally,I usually default to allowing the user to fully specify the behavior if they want.